dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageCheckOption.java

```diff
@@ -0,0 +1,103 @@
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+@AutoValue
+abstract class StaticLinkageCheckOption {
+  // TODO(suztomo): Add option to specify entry point classes
+
+  @Nullable abstract String getBomCoordinate();
+  abstract ImmutableList<String> getMavenCoordinates();
+  abstract ImmutableList<Path> getJarFileList();
+  abstract boolean isReportOnlyReachable();
+
+  static Builder builder() {
+    return new AutoValue_StaticLinkageCheckOption.Builder();
+  }
+
+  @AutoValue.Builder
+  abstract static class Builder {
+    abstract Builder setBomCoordinate(@Nullable String value);
+    abstract Builder setMavenCoordinates(List<String> value);
+    abstract Builder setJarFileList(List<Path> value);
+    abstract Builder setReportOnlyReachable(boolean value);
+    abstract StaticLinkageCheckOption build();
+  }
+
+  static StaticLinkageCheckOption parseArgument(String[] arguments) {
+    Options options = new Options();
+    options.addOption(
+        "b", "bom", true, "BOM to generate a classpath");
+    options.addOption(
+        "c", "coordinate", true, "Maven coordinates (separated by ',') to generate a classpath");
+    options.addOption("j", "jars", true, "Jar files (separated by ',') to generate a classpath");
+    options.addOption(
+        "r",
+        "report-only-reachable",
+        false,
+        "To report only linkage errors reachable from entry point");
+
+    CommandLineParser parser = new DefaultParser();
+    List<Path> jarFilePaths = new ArrayList<>();
+
+    ImmutableList.Builder<String> mavenCoordinates = ImmutableList.builder();
+    try {
+      CommandLine cmd = parser.parse(options, arguments);
+      if (cmd.hasOption("c")) {
+        String mavenCoordinatesOption = cmd.getOptionValue("c");
+        mavenCoordinates.addAll(Arrays.asList(mavenCoordinatesOption.split(",")));
+      }
+      if (cmd.hasOption("j")) {
+        String jarFiles = cmd.getOptionValue("j");
+        List<Path> jarFilesInArguments =
+            Arrays.stream(jarFiles.split(","))
+                .map(name -> (Paths.get(name)).toAbsolutePath())
+                .collect(Collectors.toList());
+        jarFilePaths.addAll(jarFilesInArguments);
+      }
+
+      String mavenBomCoordinate = cmd.getOptionValue("b");
+
+      boolean reportOnlyReachable = cmd.hasOption("r");
+
+      return builder()
+          .setBomCoordinate(mavenBomCoordinate)
+          .setMavenCoordinates(mavenCoordinates.build())
+          .setJarFileList(jarFilePaths)
+          .setReportOnlyReachable(reportOnlyReachable)
+          .build();
+    } catch (ParseException ex) {
```

Comment:
Does it make sense to just let the ParseException be thrown instead?

Reply:
I added helpFormetter.printHelp after catching ParseException.
Now ParseException is re-thrown.